File: apps/api/scripts/setup_auth.py
import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

# Путь для сохранения файла авторизации
AUTH_FILE = Path(__file__).parent.parent / "data" / "notion_auth.json"

async def setup_auth():
    print("🚀 Запуск браузера для авторизации...")
    
    async with async_playwright() as p:
        # Запускаем браузер в видимом режиме
        # Пытаемся использовать Chrome, если он установлен, иначе Chromium
        try:
            browser = await p.chromium.launch(headless=False, channel="chrome")
        except Exception:
            print("⚠️ Chrome не найден, используем стандартный Chromium...")
            browser = await p.chromium.launch(headless=False)
            
        context = await browser.new_context(
            viewport={"width": 1280, "height": 720},
            user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
        )
        page = await context.new_page()
        
        print("🔑 Переходим на страницу входа Notion...")
        await page.goto("https://www.notion.so/login")
        
        print("⏳ Пожалуйста, войдите в свой аккаунт Notion в открывшемся окне.")
        print("   Скрипт ждет, пока вы не окажетесь внутри Notion (URL будет содержать 'notion.so').")
        print("   У вас есть 3 минуты.")
        
        try:
            # Ждем, пока пользователь залогинится и URL изменится на внутренний
            # Обычно после логина перекидывает на notion.so/<workspace>
            await page.wait_for_url(lambda url: "login" not in url and "notion.so" in url, timeout=180000)
            
            print("✅ Вход обнаружен! Ждем полной загрузки...")
            # Увеличиваем таймаут ожидания загрузки сети до 60 секунд
            await page.wait_for_load_state("networkidle", timeout=60000)
            
            # Сохраняем состояние
            await context.storage_state(path=AUTH_FILE)
            print(f"💾 Сессия успешно сохранена в: {AUTH_FILE}")
            
        except Exception as e:
            print(f"❌ Ошибка или тайм-аут: {e}")
        
        await browser.close()

if __name__ == "__main__":
    asyncio.run(setup_auth())

File: apps/agent-worker/scripts/setup-auth.ts
/**
 * Скрипт для настройки авторизации Notion через браузер
 * Создает auth.json с сохраненным состоянием сессии
 */

import { chromium } from 'playwright';
import { writeFileSync, existsSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const AUTH_FILE = join(__dirname, '../../auth.json');

async function setupAuth() {
  console.log('🔐 Настройка авторизации Notion через браузер\n');
  console.log('📋 Инструкция:');
  console.log('1. Браузер откроется автоматически');
  console.log('2. Войдите в свой аккаунт Notion');
  console.log('3. После успешного входа закройте браузер');
  console.log('4. Состояние сессии будет сохранено в auth.json\n');

  const browser = await chromium.launch({
    headless: false, // Показываем браузер для входа
  });

  try {
    const context = await browser.newContext({
      viewport: { width: 1920, height: 1080 },
    });

    const page = await context.newPage();

    // Переходим на главную страницу Notion
    console.log('🌐 Открываем Notion...');
    await page.goto('https://www.notion.so', {
      waitUntil: 'networkidle',
      timeout: 30000,
    });

    console.log('\n⏳ Ожидание входа в Notion...');
    console.log('   После входа нажмите Enter в терминале...\n');

    // Ждем, пока пользователь войдет
    await new Promise<void>((resolve) => {
      process.stdin.once('data', () => {
        resolve();
      });
    });

    // Сохраняем состояние сессии
    const storageState = await context.storageState();
    writeFileSync(AUTH_FILE, JSON.stringify(storageState, null, 2));

    console.log(`\n✅ Состояние сессии сохранено в ${AUTH_FILE}`);
    console.log('   Теперь можно использовать этот файл для автоматической авторизации\n');
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    console.error(`❌ Ошибка: ${errorMessage}`);
    process.exit(1);
  } finally {
    await browser.close();
  }
}

setupAuth();
